### PR TITLE
[stable/docker-registry] Update apiVersion for Kubernetes 1.16 support

### PR DESCRIPTION
**Pull request by alexvolsh**
Signed-off-by: Alex Voloshyn <alex@geeoz.com>

#### What this PR does / why we need it:
apiVersion in the Ingress doesn't support in the Kubernetes 1.16 and above.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by alexvolsh** [comment]
@rendhalver @jpds @davidkarlsen Appreciate an ok-to-test and a quick review pleas

File: stable/docker-registry/templates/ingress.yaml [hunk 1]
```diff
@@ -2,7 +2,11 @@
 {{- $serviceName := include "docker-registry.fullname" . -}}
 {{- $servicePort := .Values.service.port -}}
 {{- $path := .Values.ingress.path -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (davidkarlsen) [inline stable/docker-registry/templates/ingress.yaml]: can maybe just do like in https://github.com/helm/charts/pull/22319/files#diff-801a58d4d0e066588b81b5f17849c124R2 ?

Comment (alexvolsh) [reply stable/docker-registry/templates/ingress.yaml]: @davidkarlsen I've changed the condition. When I try to use:
```
apiVersion: {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" -}} networking.k8s.io/v1beta1 {{- else }} extensions/v1beta1 {{- end }}
```
it fails on Go version 1.13.10

Comment (davidkarlsen) [reply stable/docker-registry/templates/ingress.yaml]: See updated PR with less chomping.

Comment (alexvolsh) [reply stable/docker-registry/templates/ingress.yaml]: @davidkarlsen thank you, I did it like in the PR

**Comment by alexvolsh** [comment]
@davidkarlsen @rendhalver should I do anything else for PR approve?

**Pull request merged**